### PR TITLE
zlib: fix MinGW clang

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/14440

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/zlib/all/conanfile.py [hunk 1]
```diff
@@ -29,7 +29,9 @@ class ZlibConan(ConanFile):
 
     @property
     def _is_mingw(self):
-        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+        return (self.settings.os == "Windows" and
+                self.settings.get_safe("compiler") in ["gcc", "clang"] and
+                self.settings.get_safe("compiler.libcxx"))
```
Comment (SpaceIm) [inline recipes/zlib/all/conanfile.py:34]:
```suggestion
                not self.settings.get_safe("compiler.runtime"))
```

File: recipes/zlib/all/conanfile.py [hunk 2]
```diff
@@ -29,7 +29,12 @@ class ZlibConan(ConanFile):
 
     @property
     def _is_mingw(self):
-        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+        host_os = self.settings.os
+        host_compiler = self.settings.get_safe("compiler")
+        mingw_gcc = host_os == "Windows" and host_compiler == "gcc"
+        mingw_clang = host_os == "Windows" and host_compiler == "clang" and \
```
Comment (prince-chrismc) [inline recipes/zlib/all/conanfile.py:35]: Could you please added a comment explaining why `not` that comes next is important for futures contributors who might not know I'd like to try avoiding breaking this again

Comment (SpaceIm) [reply recipes/zlib/all/conanfile.py:35]: Maybe just add a link to https://github.com/conan-io/conan/pull/12678

Comment (prince-chrismc) [reply recipes/zlib/all/conanfile.py:35]: Sure, "for more details  see..." it good

**Comment by mhthies** [comment]
Has this PR been abandoned on purpose? We need exactly this change to get zlib (and dependent packages like Qt, boost, etc.) working with MinGW clang at our company.

**Comment by jcar87** [comment]
> Has this PR been abandoned on purpose? We need exactly this change to get zlib (and dependent packages like Qt, boost, etc.) working with MinGW clang at our company.

Hi @mhthies  - I'm happy to pick this up again.
Unfortunately this is a result of https://github.com/conan-io/conan-center-index/pull/10771  - I suppose it may be have been to work around a Conan limitation at the time - if I left the upstream defaults everything would just work...

I'll open a PR to address this as best I can - since we can't rename libraries for already published versions.

**Pull request closed without merging**